doc/docs/master/deploy-manage/manage/upgrades.md

```diff
@@ -126,54 +116,62 @@ shows all pods as `READY`:
 
+|Secret KEY name| <div style="width:290px"> Description </div>| A - Create your secrets ahead <br> of your cluster creation| B - Pass credentials in values.yaml| <div style="width:250px"> C - Corresponding (Pachyderm Secret, Key) in which the values provided in A or B will be injected. See the following `Pachyderm Internal Secrets` section for details.</div>| 
+|------------|------------|-----|--------|---------|
+|root-token| Root clusterAdmin| pachd.rootTokenSecretName |pachd.rootToken|(pachyderm-auth, rootToken)|
+|root-token|Users give this token in their values.yaml when they install a cluster that is under an [Enterprise Server's](../../enterprise/auth/enterprise-server/setup) umbrella.|pachd.enterpriseRootTokenSecretName|pachd.enterpriseRootToken|????|
+|postgresql-password|Password to your database|global.postgresql.postgresqlExistingSecretName <br> global.postgresql.postgresqlExistingSecretKey |global.postgresql.postgresqlPassword|(postgres, postgresql-password) , (postgres, postgresql-postgres-password)|
+|OAUTH_CLIENT_SECRET|Oauth client secret for Console <br> Required if you set Console|console.config.oauthClientSecretSecretName |console.config.oauthClientSecret|(pachyderm-console-secret, OAUTH_CLIENT_SECRET)|
+|enterprise-license-key|Your enterprise license|pachd.enterpriseLicenseKeySecretName |pachd.enterpriseLicenseKey|(pachyderm-license, enterprise-license-key)|
+|pachd-oauth-client-secret| Oauth client secret for pachd| pachd.oauthClientSecretSecretName|pachd.oauthClientSecret|(pachyderm-auth, auth-config)|
+|enterprise-secret|Needed if you connect to an enterprise server|pachd.enterpriseSecretSecretName  |pachd.enterpriseSecret| (pachyderm-enterprise, enterprise-secret)  |
+|upstream-idps|The list of dex connectors, each containing Oauth client info connecting to an upstream IDP|oidc.upstreamIDPsSecretName|oidc.upstreamIDPs|(pachyderm-identity, upstream-idps)|
 
-- **C - Neither A nor B**
 
-      You have not created secrets ahead of your installation, nor did you provide values directly in the values.yaml; we created default values for you in the default secret `pachyderm-bootstrap-config` (See column C below for their key). 
-      
+### Pachyderm Internal Secrets
 
-!!! Important
-       It is important to note that if no secret name is provided for the fields mentioned in **A**, Pachyderm will retrieve the dedicated plain-text secret values in the helm values (**B**) and populate a generic, default, auto-generated secret (pachyderm-bootstrap-config) at the time of the installation (see keys in **C**). If no value is found in either one of those two cases, default values are used in `pachyderm-bootstrap-config`. This default secret is reset at each upgrade, and new default values are created causing the upgrade to fail unless they are retrieved and set back into their appropriate fields.
+In both cases (A and B above), Pachyderm reads the values provided, then injects them into a set of internal secrets. 
```

Comment:
This is maybe more of a revision comment - I think there are 3 options now: populate secrets directly, populate secrets via k8s Secrets, or DO NOTHING (sorta). The DO NOTHING option would still require today's behavior around Postgres password, but otherwise would only require the user to provide the ENTERPRISE LICENSE and optionally the IDPs if they so choose. ALL the other values will be autogenerated and upgrades should work fine. The main case we can envision for a customer to populate secrets via k8s Secrets is if they use a tool like ArgoCD for gitops because our helm templates will still include random values each time they get generated.

Does that make sense? 
cc @jrockway